### PR TITLE
change IRQ type to IRQn_Type

### PR DESCRIPTION
**Pull request by alzix**
### Description
Naively support `IRQn_Type` type in our IRQ to SPM signal mapper

### Pull request type
    [ ] Fix
    [x] Refactor
    [ ] New target
    [ ] Feature
    [ ] Breaking change

---
This change is [Reviewable](https://reviewable.io/reviews/kfnta/mbed-os/38)

### CODE REVIEW

File: tools/spm/templates/psa_NAME_partition.c.tpl [hunk 1]
```diff
@@ -95,13 +95,13 @@ static const osMutexAttr_t {{partition.name|lower}}_mutex_attr = {
 
 {% if partition.irqs|count > 0 %}
 // Mapper function from irq signal to interupts number
-uint32_t spm_{{partition.name|lower}}_signal_to_irq_mapper(uint32_t signal)
+IRQn_Type spm_{{partition.name|lower}}_signal_to_irq_mapper(uint32_t signal)
 {
     SPM_ASSERT({{partition.name|upper}}_WAIT_ANY_IRQ_MSK & signal)
```
Comment (orenc17) [inline tools/spm/templates/psa_NAME_partition.c.tpl]: add ; at the end of the line

**Comment by alzix** [comment]
Reviewed 3 of 4 files at r1, 1 of 1 files at r2.
Review status: all files reviewed, 1 unresolved discussion (waiting on @orenc17 and @danny4478)

---



*Comments from [Reviewable](https://reviewable.io/reviews/kfnta/mbed-os/38)*

**Comment by orenc17** [comment]
Reviewed 3 of 4 files at r1.
Review status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @danny4478)

---



*Comments from [Reviewable](https://reviewable.io/reviews/kfnta/mbed-os/38)*

**Comment by alzix** [comment]
@orenc17  @danny4478 ping - someone should approve it. i cannot approve to myself

**Comment by orenc17** [comment]
:lgtm:

---

Review status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @danny4478)

---



*Comments from [Reviewable](https://reviewable.io/reviews/kfnta/mbed-os/38#-:-LFpt1FBBl4cBw2-9yxT:bg7igk9)*

**Pull request merged**